Models/ProductsListPart.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Orchard.ContentManagement;
+using Orchard.ContentManagement.Utilities;
+using Orchard.Environment.Extensions;
+using Orchard.Security;
+using Newtonsoft.Json.Linq;
+
+namespace Nwazet.Commerce.Models {
+    [OrchardFeature("Nwazet.PersistentCart")]
+    public class ProductsListPart : ContentPart {
+        //This part goes in the infoset
+
+        private List<ShoppingCartItem> _items {
+            get {
+                var fromInfoset = Retrieve<string>("SerializedItems");
```

Comment:
I get that doing a first serialization to a string, and then storing that, is hacky (same stands for deserialization), but it's a simple workaround here.